compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -301,6 +302,18 @@ pub fn from_fn_attrs(cx: &CodegenCx<'ll, 'tcx>, llfn: &'ll Value, instance: ty::
             InstructionSetAttr::ArmT32 => "+thumb-mode".to_string(),
         }))
         .collect::<Vec<String>>();
+
+    // The `"wasm"` abi on wasm targets automatically enables the `+multivalue`
+    // feature because the purpose of the wasm abi is to match the WebAssembly
+    // specification, which has this feature. This won't be needed when LLVM
+    // enables this `multivalue` feature by default.
+    if cx.tcx.sess.target.arch == "wasm32" && !cx.tcx.is_closure(instance.def_id()) {
```

Comment:
This probably wants check for `is_like_wasm` now, right? As this should apply to both wasm32 and wasm64.

Stylistically it'd probably be good to merge this and the section that deals with import-module/name so that all of the wasm code in this area is in a single location.